wagtail/admin/views/pages/edit.py

```diff
@@ -375,79 +376,77 @@ def dispatch(self, request, page_id):
         return super().dispatch(request)
 
     def get(self, request):
-        if self.page_perms.user_has_lock():
-            if self.page.locked_at:
-                lock_message = format_html(
-                    _("<b>Page '{}' was locked</b> by <b>you</b> on <b>{}</b>."),
-                    self.page.get_admin_display_title(),
-                    self.page.locked_at.strftime("%d %b %Y %H:%M"),
-                )
-            else:
-                lock_message = format_html(
-                    _("<b>Page '{}' is locked</b> by <b>you</b>."),
-                    self.page.get_admin_display_title(),
-                )
+        if self.lock:
+            lock_message = None
+
+            if self.lock.reason == "locked":
```

Comment:
Should these locking reasons be defined in a class (or enum)? So we can refer to it like e.g. `Lock.Reason.LOCKED` and `Lock.Reason.WORKFLOW` (or just `LockReason.WORKFLOW`). I think that would make it easier to track down the usage of these lock reasons and avoid typos. What do you think?